### PR TITLE
feat(MangaKatana): add presence

### PR DESCRIPTION
**Pull request by shiseni**
## Description 
Add presence for [MangaKatana](https://mangakatana.com/)
* Resolves: #7336 

Settings: 
1. Multilanguage,
2. Privacy mode,
3. Show covers,
4. Show timestamps,
5. Show buttons,
6. Show the page of the manga currently reading (disabled by default).

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

| Home page | Manga's page |
| --- | --- |
| image | image |

<div align="center">Searching</div>

| By genres | By one specific Genre/Author/etc |
| --- | --- |
| image | image |

<div align="center">Reading manga</div>

| Without showing current image | With showing current image |
| --- | --- |
| image | image |

<div align="center">Privacy</div>

| Without covers, timestamps, buttons | Full privacy |
| --- | --- |
| image | image |

</details>

### CODE REVIEW

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/M/MangaKatana/presence.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		presenceData.details = privacy
+			? strings.searchSomething
+			: strings.searchFor;
+		presenceData.smallImageKey = Assets.Search;
+		presenceData.smallImageText = strings.search;
+		presenceData.state = searchInput;
+	}
+
+	if (!cover || privacy) presenceData.largeImageKey = Assets.Cover;
+	if (!buttons || privacy) delete presenceData.buttons;
```
Comment (0x796935) [inline websites/M/MangaKatana/presence.ts:180]:
```suggestion
	if ((!buttons || privacy) && presenceData.buttons) delete presenceData.buttons;
```
Adding checks if they are defined before deleting them

File: websites/M/MangaKatana/presence.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	if (!time) delete presenceData.startTimestamp;
```
Comment (0x796935) [inline websites/M/MangaKatana/presence.ts:181]:
```suggestion
	if (!time && presenceData.startTimestamp) delete presenceData.startTimestamp;
```

File: websites/M/MangaKatana/presence.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	if (privacy) delete presenceData.state;
```
Comment (0x796935) [inline websites/M/MangaKatana/presence.ts:182]:
```suggestion
	if (privacy && presenceData.state) delete presenceData.state;
```

File: websites/M/MangaKatana/presence.ts (hunk 1 above)
Comment (theusaf) [reply websites/M/MangaKatana/presence.ts:180]: I'm not sure if these are necessary. delete won't throw an error if deleting somethat that is not defined, just return false.

Comment (0x796935) [reply websites/M/MangaKatana/presence.ts:180]: They aren't necessary at all, but were recommended to me by @DarkVillager. Which made me recommend it here
https://github.com/PreMiD/Presences/pull/7346#discussion_r1189580219

Comment (DarkVillager) [reply websites/M/MangaKatana/presence.ts:180]: > I'm not sure if these are necessary. delete won't throw an error if deleting somethat that is not defined, just return false.

Agreed but it'd be better imo.

**Review by DarkVillager — APPROVED** [review]

File: websites/M/MangaKatana/presence.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+const observer = new IntersectionObserver(
+	(entries, observer) => {
+		for (const entry of entries) {
+			if (entry.isIntersecting) {
+				observer.unobserve(entry.target);
+				localStorage.setItem(
+					"currentMangaPage",
+					(<HTMLImageElement>entry.target).src
+				);
```
Comment (theusaf) [inline websites/M/MangaKatana/presence.ts:50]: You are [not allowed](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets) to access `localStorage`. Is there a reason you need to use `localStorage` over a local variable?

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**